Dev--> Master

Yaroslav7125 opened the pull request:
Баханов Ярослав
https://yaroslav7125.github.io/

Yaroslav7125 committed:
Add bad using history api

Yaroslav7125 committed:
delete  main js code

Yaroslav7125 committed:
Add ajax

Yaroslav7125 committed:
Add history api

Yaroslav7125 committed:
Fix bag with Routing

Yaroslav7125 committed:
Add pages for normal routing

Yaroslav7125 committed:
Fix bag with Routing and add bug with map

Yaroslav7125 committed:
Add loading map.

Yaroslav7125 committed:
Add normal working timer.

Yaroslav7125 committed:
Add normal loading map and add adaptation for mapPage and timerPage

Yaroslav7125 committed:
rewrote css for a in .header-part

Yaroslav7125 committed:
rewrote css for a in header and rewrote js

Yaroslav7125 committed:
Merge branch 'feature/CreatingSPA' into dev

crutch12 commented:
Пункты меню должны подсвечиваться в зависимости от текущей страницы

crutch12 commented:
![image](https://user-images.githubusercontent.com/19373212/126540191-6b6a8837-6b4d-4f97-be61-0528c5098240.png)
Если открыть сайт не на карте, то ошибка в консоли
UPD: в любом случае ошибка. Не должно быть

Yaroslav7125 committed:
fixed error from console

Yaroslav7125 committed:
Add backlight for menu.

Yaroslav7125 commented:
> Пункты меню должны подсвечиваться в зависимости от текущей страницы

Пункты меню подсвечиваются. Решение в коммите 8f0636b.

Yaroslav7125 commented:
> ![image](https://user-images.githubusercontent.com/19373212/126540191-6b6a8837-6b4d-4f97-be61-0528c5098240.png)
> Если открыть сайт не на карте, то ошибка в консоли
> UPD: в любом случае ошибка. Не должно быть

Исправил код выводящий ошибку в консоль, больше он её туда не выводит, решение в коммите 48514a8

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+function getPath(str){
+    return str.split('/')[str.split('/').length-1];
+}
+divElem = document.querySelector(".nav"); // navbar в котором находятся ссылки
```
не используется

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+    return str.split('/')[str.split('/').length-1];
+}
+divElem = document.querySelector(".nav"); // navbar в котором находятся ссылки
+content = document.getElementById('content') // div элемент куда вставляется контент
```
у переменных нужно указывать тип переменной (let/var/const)
желательно всегда использовать const, если её не надо менять

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+divElem = document.querySelector(".nav"); // navbar в котором находятся ссылки
+content = document.getElementById('content') // div элемент куда вставляется контент
+const ShowContent = data => {
+    content.innerHTML = data;
```
я бы тут лучше getElementById вызывал, вдруг на старте его нету

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+}
+divElem = document.querySelector(".nav"); // navbar в котором находятся ссылки
+content = document.getElementById('content') // div элемент куда вставляется контент
+const ShowContent = data => {
```
функции с маленькой буквы

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+    }
+}
+let flag = false;
+function GetContent(ThePath){
```
аргументы с маленькой буквы
ну и лучше назвать это href/link/path

Yaroslav7125 replied:
переименовал, учту на будущее)

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+            });
+            break;
+        case '/activity.html':
+            //history.pushState(null, null, ThePath)
```
зачем коммент оставил?

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+function GetContent(ThePath){
+    switch(ThePath){
+         case '/':
+            makeGetRequest(`${location.origin}/templates${ThePath}activity.html`, (data) => ShowContent(data));
```
ThePath в строке здесь выглядит странно... лучше бы просто / написал

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+    if (timer == null)return;
+    let secStr = String(sec%60).padStart(2, '0');
+    let hoursStr = String(Math.trunc(sec/3600)).padStart(2, '0');
+    let minutesStr = String(Math.trunc(sec/60)).padStart(2, '0');
```
поменяй порядок с hoursStr то...

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+    let minutesStr = String(Math.trunc(sec/60)).padStart(2, '0');
+    timer.innerHTML =  `${hoursStr}:${minutesStr}:${secStr}`;   
+}
+function Timer(){
```
`startTimer`

Yaroslav7125 replied:
Переименовал Timer --> startTimer

crutch12 commented on index.js:
```diff
@@ -0,0 +1,89 @@
+    showTime();
+}
+document.addEventListener("DOMContentLoaded", Timer);
+showTime(); // при переходе должен быть запущен таймер
```
Вынеси вызов этой функции в Timer, зачем её вызывать до DOMContentLoaded?

Yaroslav7125 replied:
Удалил 89 строку так-как функция showTime вызывается в countTime.

crutch12 reviewed (commented):
Оставил комменты

Yaroslav7125 committed:
Deleted getPath

Yaroslav7125 committed:
Corrected  'divElem' and 'content'

Yaroslav7125 committed:
Переименовал функцию ShowContent --> showContent.
Исправил строчку "content = document.getElementById('content');"
Внёс её в функцию showContent();

Yaroslav7125 committed:
Переименовал аргумент функции на path. Удалил коммент.

Yaroslav7125 committed:
Сменил путь при роутинге на корень.

Yaroslav7125 committed:
Поменял порядок с hoursStr.

Yaroslav7125 committed:
Переименовал Timer в более подходящее startTimer.

Yaroslav7125 committed:
Удалил showTime, так как она вызывается в countTime.

Yaroslav7125 committed:
Переименовал GetContent --> getContent

crutch12 commented:
Congratulations, ёпта
